### ISSUE TITLE
wasmJs support for multiplatform-settings-coroutines

### ISSUE THREAD

**Issue by ahna92** [russhwolf/multiplatform-settings#188]
As wasm went stable, need support for `multiplatform-settings-coroutines` ,

Also [kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines/releases/tag/1.8.0) supported wasm which was the blocker i think 

Tired to support in https://github.com/russhwolf/multiplatform-settings/pull/187 , but i feel some configurations in `BuildHelpers.kt` blocked it

**Comment by russhwolf** [comment]
Wasm is not stable. But it is stable enough for kotlinx to include it, which is good enough for here.

Unfortunately, Turbine is also a blocker here.

There are some refactors in the 1.2 branch because the old BuildHelpers config is terrible. This will be trivial to add on top of the existing changes in 1.2.

**Comment by ahna92** [comment]
Great, thanks !

**Comment by armond1978** [comment]
@ahna92 Is there any ETA for v1.2 and WASM support?

**Comment by armond1978** [comment]
@russhwolf Is there any ETA for v1.2 and WASM support?

**Comment by shriharsha-srb** [comment]
Yes would be nice to know if this is in progress :)

**Comment by shriharsha-srb** [comment]
@russhwolf  Would be nice to know if this is in progress and if its gonna be released with 1.2 version. Thanks :)

**Comment by xxfast** [comment]
Turbine 1.1.0 now has wasmJs target

**Comment by russhwolf** [comment]
Been traveling and otherwise busy, but my current goal is to get this out as part of 1.2 by the end of the month.

**Comment by shriharsha-srb** [comment]
Are there any branches/snapshot versions where I can try support for wasmJs ?

**Comment by shriharsha-srb** [comment]
@russhwolf Can you plz provide some updates here ? Also let me know if some snapshot version if available. That works as well.

**Comment by burnoo** [comment]
@shriharsha-srb I've published snapshot of 1.2.0 to [my public maven repository](https://github.com/burnoo/maven). Feel free to use it while waiting for official release ☺️

#### Setup
Add maven In `settings.gradle.kts` (read more about versioning [here](https://github.com/burnoo/maven?tab=readme-ov-file#what-is--beapn))
```kotlin
dependencyResolutionManagement {
    repositories {
        maven(url = "https://pkgs.dev.azure.com/burnoo/maven/_packaging/public/maven/v1") {
            content {
                includeVersionByRegex(".*", ".*", ".*-beap[0-9]+")
            }
        }
    }
}
```
In your module:
```kotlin
implementation("com.russhwolf:multiplatform-settings:1.2.0-beap1")
```

**Comment by stephanepechard** [comment]
@burnoo Thanks for releasing this!

Unfortunately, it does not work at the moment, I always get:
`Could not find com.russhwolf:multiplatform-settings:1.2.0-beap1.`
Is it still available?

**Comment by burnoo** [comment]
@stephanepechard 

Yes it's still available. Have you added my maven to `settings.gradle.kts`? You can check out similar configuration [here](https://github.com/burnoo/compose-remember-setting/blob/main/settings.gradle.kts)

**Comment by stephanepechard** [comment]
@burnoo 
Thanks for your answer.
I have copied the exact same config, with no much luck :-(

<img width="904" alt="Screenshot 2024-08-21 at 18 17 13" src="https://github.com/user-attachments/assets/2db7b2a1-96d9-44a2-b21c-0fd9e22a0126">

**Comment by burnoo** [comment]
I've just checked it in my local environment and maven works as expected. Could you show the rest of the file and `implementation` usage?

**Comment by stephanepechard** [comment]
@burnoo Sure, but maybe not on this thread, can I write you an email?
Thanks!

**Comment by kit0kat** [comment]
Hi @burnoo thanks for publishing 1.2.0 !

I just added it to my project and tried to use it on wasmJs:

`val settings = StorageSettings(localStorage).toFlowSettings()`

But `.toFlowSettings()` does not work since `StorageSettings` does not inherit from `ObservableSettings` like the Android and iOS implementation. 

This works on Android and iOS:
`val settings = SharedPreferencesSettings(sharedPreferences).toFlowSettings()`
`val settings = NSUserDefaultsSettings(NSUserDefaults.standardUserDefaults()).toFlowSettings()`


Is this already supported in 1.2.0 ? 

@russhwolf Will there be an official release soon?

**Comment by burnoo** [comment]
Hi @kit0kat. `1.2.0` comes with the new artifact `com.russhwolf:multiplatform-settings-make-observable` that allows to observe `StorageSettings` at runtime. Important thing to mention is that it will only trigger updates for the current `ObservableSettings` instance ([docs](https://github.com/russhwolf/multiplatform-settings/blob/1.2/multiplatform-settings-make-observable/src/commonMain/kotlin/MakeObservableSettings.kt#L36)).

Artifact is published to my maven, so you can do the following
1. Add `com.russhwolf:multiplatform-settings-make-observable:1.2.0-beap1` to dependencies
2. Use `StorageSettings(localStorage).makeObservable().toFlowSettings()`

**Comment by kit0kat** [comment]
@burnoo Thank you! Now it works as expected. Many thanks!

**Comment by russhwolf** [comment]
Sorry it took so long. This has now been released in version 1.2.0.
